### ISSUE TITLE
Geyser renders Overworld as nether when datapacks that add dimensions are used

### ISSUE THREAD

**Issue by rnope-idk-how-to-code** [GeyserMC/Geyser#4837]
### Describe the bug

Hello! We’ve run into multiple problems on our server running fabric geyser, we have multiple datapacks but the main ones are Bracken Pack and Terralith. 
Server is quite powerful and Java players don’t encounter any lags or problems at all. 
However bedrock experience is close to unplayable, to start with for bedrock players it is ALWAYS day, weather and day circle don’t seem to change at all. 
The second problem is world is cut in half and chunks don’t seem to load beneath certain level (I don’t remember this being a problem previously) in console it says:
`[21:25:30] [Geyser player thread-5-1/WARN]: The server is using chunk heights bigger than Bedrock can accept! To avoid problems, chunk height must remain between Y0 and Y128 for dimension 3.`

However “dimension 3” is over world, I don’t think think this is how it’s supposed to work

tl;dr When datapacks that add dimensions are introduced overworld is rendered as nether

Datapack which is most likely a reason for such behaviour is Bracken pack https://modrinth.com/datapack/the-bracken-pack
And maybe Terralith also plays into that somehow

### To Reproduce

1. run fabric server with geyser on 1.21
2. Upload world file with following datapacks: The Bracken Pack, Terralith
3. Run server, now geyser renders overworld as nether for whatever reason (underwater is red) fog is there, and now daylight cycle or weather

### Expected behaviour

From my previous testing with Bracken Pack I expected that overworld would function as normal and not as nether

Dimensions from Bracken Pack are limited to height of nether (and this limitation is understandable) 

### Screenshots / Videos

![Screenshot_20240707_133129_Minecraft](https://github.com/GeyserMC/Geyser/assets/172921659/96d54a87-1a83-4a11-9536-bdb3bcbcbdec)
![Screenshot_20240706_223851_Minecraft](https://github.com/GeyserMC/Geyser/assets/172921659/3303b7e2-2ec8-4914-9fea-b4c5f1e32a78)

This is in overworld btw

### Server Version and Plugins

Server is running on Fabric 1.21

I’m mostly certain mods aren’t causing this bug yet here they are:

BetterPingDisplay-Fabric-1.21-1.1.1.jar
calcmod-1.3.2+fabric.1.21.jar
cloth-config-15.0.127-fabric.jar
collective-1.21.0-7.70.jar
darcenos-minecarts-1.3.jar
deathbackup-1.21.0-3.3.jar
Discord-MC-Chat-2.3.3.jar
drogstyle-1.3.4+1.21.jar
fabric-api-0.100.4+1.21.jar
fabric-language-kotlin-1.11.0+kotlin.2.0.0.jar
FallingTree-1.21-1.21.0.5.jar
ferritecore-7.0.0-fabric.jar
Floodgate-Fabric-2.2.3-SNAPSHOTbuild.28.jar
fsit-2.4.1+1.21.jar
furnacerecycle-1.21.0-2.4.jar
geyser-fabric-2.4.0-SNAPSHOT+build.595.jar
grassseeds-1.21.0-3.2.jar
htm-1.1.13.jar
hurricane-neoforge-2.0.0+build.179.jar
jsmt-2.0_1.21.jar
ledger-1.3.4.jar
lithium-fabric-mc1.21-0.12.7.jar
memoryleakfix-fabric-1.17+-1.1.5.jar
modernfix-fabric-5.18.3+mc1.21.jar
nohostilesaroundcampfire-1.21.0-7.0.jar
sodium-extra-0.5.7+mc1.21.jar
sodium-fabric-0.5.11+mc1.21.jar
styled-nicknames-1.6.0+1.21.jar
styledplayerlist-3.5.1+1.21.jar
vanilla-refresh-1.4.20b.jar
vcinteraction-fabric-1.21-1.0.7.jar
vmp-fabric-mc1.21-0.2.0+beta.7.162-all.jar
voicechat-fabric-1.21-2.5.17.jar

### Geyser Dump

https://dump.geysermc.org/Uhldd0p3HfzITHX7yn7YU5LSVLyrDQEd

### Geyser Version

2.4.0-SNAPSHOT-576

### Minecraft: Bedrock Edition Device/Version

1.21

**Comment by rnope-idk-how-to-code** [comment]
I've been told that hurricane version is wrong, to test this I've removed hurricane yet issue sadly remains

**Comment by YHDiamond** [comment]
I was able to replicate both the daylight issue and the dimension issue with just Paper 1.21, standalone Geyser, and the bracken pack. 
I want to note that from Java, the new dimension (accessed by going above build height limit) visibly has an unchanging dark night with the moon in the sky, but using `/time query` on Java and `/gamerule doDaylightCycle` I got that the time was constantly changing and going up even though I did not see it changing in the sky. Bedrock is properly interpreting the time given to it, it just somehow is that it's not time that is dictating the sky but something else.

**Comment by Camotoy** [comment]
The builds here https://github.com/GeyserMC/Geyser/actions/runs/9922258255 should fix the issue, but they do need testing.

**Comment by yomli** [comment]
I had a similar issue where a custom dimension caused the overworld sky to not render properly, in a tv noise way. As a workaround, I disabled the _Beautiful Skies_ option in the _Video Settings_. But the build for Fabric fixed the issue indeed.
